### PR TITLE
chore: add SECURITY.md

### LINKED ISSUE #13276: docs: Security reporters need a clear reporting policies document such as SECURITY.md
Deno is becoming popular as a secure runtime in the world. However, the procedure for reporting issues leading to vulnerabilities remains unclear. 
Security engineers who discover a vulnerability will first check `SECURITY.md`,  so just having this will help. 
In addition, the limitations on Security Guarantee should be written as clear as possible in security policies. 
There are many discussions whether they are vulnerabilities or not, because many "secure" runtime is not clear at which layer security is guaranteed. 
For example, if all the code is coded by a malicious user, does Deno continue to guarantee the safety of the sandbox? 
rel  #8094  #12058

### PR DESCRIPTION
**Pull request by lucacasonato**
Closes #13276

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+The following paragraphs outline the rough security model for Deno. The model
+may change slightly over time, but in general the model is as follows:
+
+- All JavaScript run in Deno is considered untrusted. Permissions are thus never
+  enforced in JavaScript, as all JavaScript is considered untrusted
+- All JavaScript run in a single Deno process is considered to be part of the
+  same program and is not isolated from itself. This means that Deno does not
+  guarantee that values set by one JS module will be inaccessible to another, or
+  that a value set in one web worker can not be accessed by another.
+- All runtime I/O is considered to be privledged and must always be guarded by a
```
Comment (littledivy) [inline SECURITY.md:50]:
```suggestion
- All runtime I/O is considered to be privileged and must always be guarded by a
```

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Security Policy
+
+Thank you for taking the time to investigate the security of Deno. The security
+of Deno is our top most priority. appreciate investigative work into system
```
Comment (littledivy) [inline SECURITY.md:4]:
```suggestion
of Deno is our top most priority. We appreciate investigative work into system
```
maybe?

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- We will handle your report with strict confidentiality, and not pass on your
+  personal details to third parties without your permission.
+
+## Security Model
+
+The following paragraphs outline the rough security model for Deno. The model
+may change slightly over time, but in general the model is as follows:
+
+- All JavaScript run in Deno is considered untrusted. Permissions are thus never
+  enforced in JavaScript, as all JavaScript is considered untrusted
```
Comment (satyarohith) [inline SECURITY.md:45]:
```suggestion
- All JavaScript run in Deno is considered untrusted, so permissions are thus never
  enforced in JavaScript.
```

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  guarantee that values set by one JS module will be inaccessible to another, or
+  that a value set in one web worker can not be accessed by another.
+- All runtime I/O is considered to be privileged and must always be guarded by a
+  runtime permission. This includes filesystem access, network access, etc.
+- Users should not be able to self-escalate their permissions without explicit
+  consent.
+- I/O required to build an initial static module graph should always follow the
+  permissions of it's parent. If there is no parent, all permissions are
+  granted. As an example, this means that the initial static module graph that
+  is construced when doing `deno run`, does not have any permission
```
Comment (satyarohith) [inline SECURITY.md:57]:
```suggestion
- I/O required to build an initial static module graph should always follow the
  permissions of its parent. If there is no parent, all permissions are
  granted. As an example, this means that the initial static module graph that
  is constructed when doing `deno run`, does not have any permission
```

File: SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+**Please do not open security issues in the public issue tracker.**
+
+### Please do the following
+
+- Do not take advantage of the vulnerability or problem you have discovered.
+- Do not publish or reveal the problem until it has been resolved.
+- Do not use attacks on physical security or applications of third parties.
+- Do provide sufficient information to reproduce the problem, so we will be able
+  to resolve it as quickly as possible. Usually a list of steps to follow, and
```
Comment (satyarohith) [inline SECURITY.md:28]:
```suggestion
  to resolve it as quickly as possible. Usually, a list of steps to follow, and
```

File: SECURITY.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# Security Policy
+
+Thank you for taking the time to investigate the security of Deno. The security
+of Deno is our top most priority. We appreciate investigative work into system
+security carried out by well-intentioned, ethical security researchers. If you
+discover a vulnerability, however small, we would like to know about it so we
+can address it with appropriate measures, as quickly as possible. This document
+outlines the method we use to work with the security research community to
+address runtime security.
```
Comment (satyarohith) [inline SECURITY.md:9]:
```suggestion
Thank you for taking the time to investigate the security of Deno. The security
of Deno is our topmost priority. We appreciate investigative work into system
security by well-intentioned, ethical security researchers. If you
discover a vulnerability, however small, we would like to know about it to address it with appropriate measures as quickly as possible. This document
outlines the method we use to work with the security research community to
address runtime security.
```

**Review by satyarohith — COMMENTED** [review]
few nits

**Review by ry — APPROVED** [review]
LGTM

File: SECURITY.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+- Users should not be able to self-escalate their permissions without explicit
+  consent.
+- I/O required to build an initial static module graph should always follow the
+  permissions of its parent. If there is no parent, all permissions are granted.
+  As an example, this means that the initial static module graph that is
+  constructed when doing `deno run`, does not have any permission restrictions.
+  However, the module graph constructed as the result of loading a web worker or
+  dynamic import will be restricted to the permissions of the caller (the main
+  worker most likely).
+- We consider the V8 VM to be a secure sandbox.
```
Comment (bnoordhuis) [inline SECURITY.md:60]: Belated drive-by comment: could perhaps use some explication on whether that means we treat bugs in V8 as security issues or as not-our-problems.

(I assume the latter because ultimately they're not under our control.)
